contracts/extensions/FeeTaker.sol

```diff
@@ -182,6 +173,29 @@ contract FeeTaker is IPostInteraction, AmountGetterWithFee, Ownable {
         return _isWhitelistedGetterImpl(whitelistData, taker);
     }
 
+    /**
+     * @dev Calculates fee amounts depending on whether the taker is in the whitelist and whether they have an _ACCESS_TOKEN.
+     * `extraData` consists of:
+     * 2 bytes — integrator fee percentage (in 1e5)
+     * 1 bytes - integrator rev share percentage (in 1e2)
+     * 2 bytes — resolver fee percentage (in 1e5)
+     * bytes — whitelist structure determined by `_isWhitelistedPostInteractionImpl` implementation
+     * Override this function if the calculation of integratorFee and protocolFee differs from the existing logic and requires a different parsing of extraData.
+     */
+    function _getFeeAmounts(IOrderMixin.Order calldata /* order */, address taker, uint256 takingAmount, uint256 /* makingAmount */, bytes calldata extraData) internal virtual returns (uint256 integratorFeeAmount, uint256 protocolFeeAmount, bytes calldata tail) {
+        bool isWhitelisted;
+        uint256 integratorFee;
+        uint256 integratorShare;
+        uint256 resolverFee;
+        (isWhitelisted, integratorFee, integratorShare, resolverFee, tail) = _parseFeeData(extraData, taker, _isWhitelistedPostInteractionImpl);
```

Comment:
We can reduce the number of lines without increasing gas costs
```
(bool isWhitelisted, uint256 integratorFee, uint256 integratorShare, uint256 resolverFee, bytes calldata parsedTail) = _parseFeeData(extraData, taker, _isWhitelistedPostInteractionImpl);
tail = parsedTail;
```

Reply:
Fixed in 6b6346aa4e72cebf9c02584c894ac0dd984ea516